pkg/kubectl/cmd/testing/fake.go

```diff
@@ -515,7 +516,11 @@ func (f *fakeAPIFactory) Decoder(bool) runtime.Decoder {
 }
 
 func (f *fakeAPIFactory) JSONEncoder() runtime.Encoder {
-	return testapi.Default.Codec()
+	if f.tf.JSONEncoder != nil {
```

Comment:
I need to use testapi.Extension.Codec() for the encoder. By default the fakeAPIFactory uses the testapi.Default.Codec and that causes this error 

```
=== RUN   TestApplyTPR
Path: /thirdpartyresources/test-tpr.unit-test.test.com, Method: GET--- FAIL: TestApplyTPR (0.01s)
	cmd_test.go:50: Error running command (exit code 1): error: error when applying patch:

		to:
		&{0xc4204de0e0 0xc4204debd0  test-tpr.unit-test.test.com ../../../test/fixtures/pkg/kubectl/cmd/apply/third-party-resource.yaml &ThirdPartyResource{ObjectMeta:k8s_io_kubernetes_pkg_api_v1.ObjectMeta{Name:test-tpr.unit-test.test.com,GenerateName:,Namespace:,SelfLink:,UID:,ResourceVersion:,Generation:0,CreationTimestamp:0001-01-01 00:00:00 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{kubectl.kubernetes.io/last-applied-configuration: ,},OwnerReferences:[],Finalizers:[],ClusterName:,},Description:A TPR definition used for unit tests,Versions:[{v1}],} &TypeMeta{Kind:,APIVersion:,}  false}
		for: "../../../test/fixtures/pkg/kubectl/cmd/apply/third-party-resource.yaml": error when serializing current configuration from:
		&TypeMeta{Kind:,APIVersion:,}
		for: "../../../test/fixtures/pkg/kubectl/cmd/apply/third-party-resource.yaml": extensions.ThirdPartyResource is not suitable for converting to ["v1"]
FAIL
exit status 1
FAIL	k8s.io/kubernetes/pkg/kubectl/cmd	0.039s
```